### ISSUE TITLE
JSON report format breaking changes in 4.6.2

### ISSUE THREAD

**Issue by martincostello** [dotnet-outdated/dotnet-outdated#554]
The changes in #545 have changed the format of the JSON report, which is breaking automation I have that ingests the content of the file.

A before an example report is shown below.

The main breaking changes are that `UpgradeSeverity` has changed from a string to a number and `Name` has changed from a string to an object.

There's also a lot more content, but I don't know if that's intended or not.

### Report from 4.6.1

```json
{
  "Projects": [
    {
      "Name": "TweetiePie",
      "FilePath": "C:\\Coding\\codeflow\\tweetiepie\\src\\TweetiePie\\TweetiePie.csproj",
      "TargetFrameworks": [
        {
          "Name": "net9.0",
          "Dependencies": [
            {
              "Name": "Microsoft.Extensions.Http.Diagnostics",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": "Major"
            },
            {
              "Name": "Microsoft.Extensions.Http.Resilience",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": "Major"
            },
            {
              "Name": "Microsoft.Extensions.Telemetry",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": "Major"
            }
          ]
        }
      ]
    }
  ]
}
```

### Report from 4.6.2

```json
{
  "Projects": [
    {
      "TargetFrameworks": [
        {
          "Dependencies": [
            {
              "Description": "Microsoft.Extensions.Http.Diagnostics",
              "IsAutoReferenced": false,
              "IsTransitive": false,
              "IsVersionCentrallyManaged": true,
              "Name": "Microsoft.Extensions.Http.Diagnostics",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": 3
            },
            {
              "Description": "Microsoft.Extensions.Http.Resilience",
              "IsAutoReferenced": false,
              "IsTransitive": false,
              "IsVersionCentrallyManaged": true,
              "Name": "Microsoft.Extensions.Http.Resilience",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": 3
            },
            {
              "Description": "Microsoft.Extensions.Telemetry",
              "IsAutoReferenced": false,
              "IsTransitive": false,
              "IsVersionCentrallyManaged": true,
              "Name": "Microsoft.Extensions.Telemetry",
              "ResolvedVersion": "9.0.0-preview.4.24261.1",
              "LatestVersion": "9.0.0-preview.4.24263.5",
              "UpgradeSeverity": 3
            }
          ],
          "Name": {
            "Fallback": [
              {
                "Framework": ".NETFramework",
                "Version": "4.6.1.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.6.1",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.6.2.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.6.2",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.7.0.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.7",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.7.1.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.7.1",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.7.2.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.7.2",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.8.0.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.8",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              },
              {
                "Framework": ".NETFramework",
                "Version": "4.8.1.0",
                "Platform": "",
                "PlatformVersion": "0.0.0.0",
                "HasPlatform": false,
                "HasProfile": false,
                "Profile": "",
                "DotNetFrameworkName": ".NETFramework,Version=v4.8.1",
                "DotNetPlatformName": "",
                "IsPCL": false,
                "IsPackageBased": false,
                "AllFrameworkVersions": false,
                "IsUnsupported": false,
                "IsAgnostic": false,
                "IsAny": false,
                "IsSpecificFramework": true
              }
            ],
            "RootFramework": {
              "Framework": ".NETCoreApp",
              "Version": "9.0.0.0",
              "Platform": "",
              "PlatformVersion": "0.0.0.0",
              "HasPlatform": false,
              "HasProfile": false,
              "Profile": "",
              "DotNetFrameworkName": ".NETCoreApp,Version=v9.0",
              "DotNetPlatformName": "",
              "IsPCL": false,
              "IsPackageBased": true,
              "AllFrameworkVersions": false,
              "IsUnsupported": false,
              "IsAgnostic": false,
              "IsAny": false,
              "IsSpecificFramework": true
            },
            "Framework": ".NETCoreApp",
            "Version": "9.0.0.0",
            "Platform": "",
            "PlatformVersion": "0.0.0.0",
            "HasPlatform": false,
            "HasProfile": false,
            "Profile": "",
            "DotNetFrameworkName": ".NETCoreApp,Version=v9.0",
            "DotNetPlatformName": "",
            "IsPCL": false,
            "IsPackageBased": true,
            "AllFrameworkVersions": false,
            "IsUnsupported": false,
            "IsAgnostic": false,
            "IsAny": false,
            "IsSpecificFramework": true
          }
        }
      ],
      "Name": "C:\\Coding\\codeflow\\tweetiepie\\src\\TweetiePie\\TweetiePie.csproj",
      "FilePath": "C:\\Coding\\codeflow\\tweetiepie\\src\\TweetiePie\\TweetiePie.csproj"
    }
  ]
}
```

**Comment by martincostello** [comment]
4.6.3 fixes most of the issues. The only remaining issue appears to be that `Name` has changed to the value of `FullPath`.

**Comment by slang25** [comment]
Fixed in 4.6.4
